client-libraries/rust/src/plugin_utils/mod.rs

```diff
@@ -1,9 +1,13 @@
 //! Various helpful utilities for writing modality-reflector plugins.
 
 pub mod config;
+pub mod serde;
+
+#[cfg(feature = "modality")]
```

Comment:
Can we add a build-with-only-modality-feature step to the CI job?

Just gating the modules here will still break modality-feature-only builds since there are `mutation` types used in `src/plugin_utils/config.rs` that aren't gated by anything.